index.d.ts

```diff
@@ -437,4 +438,8 @@ type StringValue = string | void | null;
 	interface IPluginMiddleware<T> extends IPlugin<T> {
 		register_middlewares(app: any, auth: IBasicAuth<T>, storage: IStorageManager<T>): void;
 	}
+
+	interface IPluginStorageFilter<T> extends IPlugin<T> {
+		filter(packageInfo: Package, cb: Callback): void;
```

Comment:
Here you are filtering `metadata` only. But the method name is too generic, perhaps someone wanna filter `distTags`, `versions`, or avoid `removePackages` in the future. I think this should be more specific to the task is executing, in this case filtering metadata.

Reply:
`filter_metadata`?

Reply:
go ahead, LGTM